### PR TITLE
Add linter

### PR DESCRIPTION
**Pull request by jarrodmillman**
Not sure folks are interested in this, but pre-commit has been pretty helpful for the other projects I work on. This does introduce some linting noise, but it is ignored by git and GitHub. I am happy to close this or modify it as folks see fit.

If you don't want to add this via pre-commit, I could make a separate PR that just updates the source files with pyupgrade and flynt to remove a ton of older Python cruft.

Thoughts?

### CODE REVIEW

**Comment by rgommers** [comment]
This seems like a useful set of changes to me. And I'm fine with adding the pre-commit yaml file. It doesn't seem to be triggering bot PRs, and the selected linter option seem okay (adding too many is a problem for usability, the ones here are okay with me).

I'll merge this soon unless there are any concerns. I haven't reviewed the full diff yet, but I assume they're all valid changes since they-re tool generated.

**Comment by jarrodmillman** [comment]
Just a reminder: This will need to be merged with a "Create a merge commit" for the `.git-blame-ignore-revs` commits to be valid.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11"]
+
+    steps:
+      - uses: actions/checkout@v3
```
Comment (cclauss) [inline .github/workflows/lint.yml:17]:
```suggestion
      - uses: actions/checkout@v4
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: fc260393cc4ec09f8fc0a5ba4437f481c8b55dc1 # frozen: v3.0.3
+    hooks:
+      - id: prettier
+        files: \.(md|rst|toml|yml|yaml)
+        args: [--prose-wrap=preserve]
+
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: 36c161038ec6d1447949edd5c0060436cb483c9b # frozen: v0.0.289
```
Comment (cclauss) [inline .pre-commit-config.yaml:24]: v0.0.290

**Comment by rgommers** [comment]
Would you mind fixing the merge conflicts?

**Comment by jarrodmillman** [comment]
@rgommers We decided to make pywavelets an optional dependency of scikit-image so we could get 3.12 wheels out:
https://pypi.org/project/scikit-image/0.22.0rc1/

But it would still be nice to get 3.12 wheels for pywt out soon. Let me know if I can help. I am happy to do release management for the next release if it would be helpful.

**Comment by jarrodmillman** [comment]
@rgommers Any updates on this or a new release? I am happy to handle the release if it would help. Thanks!!

**Comment by jarrodmillman** [comment]
@rgommers It would be great to get 3.12 binaries out soon. This is the only scikit-image (optional) dependency that doesn't support Python 3.12.

**Comment by rgommers** [comment]
Apologies for the delay @jarrodmillman . New release is up now. I'll deal with remaining PR reviews that were not needed for 1.5.0, including this one, afterwards (hopefully tomorrow).

**Review by rgommers — APPROVED** [review]
LGTM. I went through everything now - most of the diff is reordering of imports, moving to f-strings, and minor style changes. There were only a couple of lines to check more carefully; all looked fine and CI is green.

The new CI job required a bump and a small fix for other new code that was merged after this PR was opened. I don't expect it to be a burden; if it fails it should be obvious from the CI log what needs fixing. Development velocity on this repo is pretty low though; if the new lint job fails too much on average it can be disabled in the future (I expect it's fine though, since 5 months later only a few tweaks were needed).

In it goes! Thanks again Jarrod.

**Pull request merged**